internal/npm_install/process_pnp_file.ts

```diff
@@ -0,0 +1,23 @@
+import * as fs from "fs";
+import * as process from "process";
+import * as path from "path";
+if (require.main === module) {
+  main();
+}
+
+/**
+ * This patches a PnP file to force the resolver to resolve from the repository's directory
+ * not the entry_point's directory. PnP assumes they are the same so when bazel creates the
+ * pnp file in a different directory, the pnp files can't resolve the paths.
+ */
+function main() {
+  const workspacePath = process.argv.slice(2)[0];
+
+  const pnpPath = path.join(workspacePath, ".pnp.js");
+  const pnpFile = fs.readFileSync(pnpPath, "utf8");
+  const patchedPnpFile = pnpFile.replace(
```

Comment:
can you add some docs here with an example of what this is supposed to match? Also, do we expect to only replace the first mach?

Reply:
we expect their to only ever be 1 match, this is a hack to force the path we look for PnP modules to be the same as where `yarn_install` is invoked. If we don't do this the pnp resolver defaults to the path the nodejs binary is invoked, which for bazel is different.

I suppose it's not necessarily only supposed to replace once, so we should do a global replace to make it better